hydra-cluster/src/Hydra/Cluster/Faucet.hs

```diff
@@ -106,19 +106,30 @@ returnFundsToFaucet ::
   RunningNode ->
   Actor ->
   IO ()
-returnFundsToFaucet tracer RunningNode{networkId, nodeSocket} sender = do
+returnFundsToFaucet tracer node sender = do
   (faucetVk, _) <- keysFor Faucet
-  let faucetAddress = mkVkAddress networkId faucetVk
+  senderKeys <- keysFor sender
+  returnAmount <- returnFundsToFaucet' tracer node faucetVk senderKeys
+  traceWith tracer $ ReturnedFunds{actor = actorName sender, returnAmount}
 
-  (senderVk, senderSk) <- keysFor sender
+returnFundsToFaucet' ::
+  Tracer IO FaucetLog ->
+  RunningNode ->
+  VerificationKey PaymentKey ->
+  (VerificationKey PaymentKey, SigningKey PaymentKey) ->
```

Comment:
Could: only pass `SigningKey PaymentKey`

you can always get the verification key from the signing key, while we always require the signing key. This should make call sites simpler